include/Parser/Parser.h

```diff
@@ -10,6 +10,18 @@
 
 namespace parser {
 
+/// Custom error class representing any syntax errors encountered during parsing.
+/// The parser terminates and returns an empty AST on the first syntax error it encounters.
+class SyntaxError {
```

Comment:
Is defining the whole SyntaxError class in the header like this acceptable from a style perspective? It seemed unnecessary to put just the definition of getMessage() in Parser.cpp.

Reply:
I like the idea of a `SyntaxError` type. Because parsing will happen in different "contexts" (e.g. as part of a build, in the language server for IDE support, etc.), perhaps we want to allow consumers of the parser output to define the effect of the syntax error — they would receive a `SyntaxError` value (or maybe a collection of them, someday), and could write it to stdout, draw an error squiggle, etc.

If we go that route, this class should also represent the location and length of the troublesome token. That's potentially beyond the scope of this bug and PR, though.